lib/administrate/field/has_one.rb

```diff
@@ -3,17 +3,25 @@
 module Administrate
   module Field
     class HasOne < Associative
-      def self.permitted_attribute(attr, options = nil)
-        associated_class_name =
-          if options
-            options.fetch(:class_name, attr.to_s.singularize.camelcase)
+      def self.permitted_attribute(attr, options = {})
+        resource_class = options[:resource_class]
+        _associated_class_name =
```

Comment:
why the leading underscore?

Reply:
To distinguish it from the method `Associative.associated_class_name`, which is in fact used below in line 13.

I agree it's confusing, I have just renamed it `final_associated_class_name`, which may work better.